packages/bin-etl/src/index.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs'
+
+const fileName = './worlds.json'
+processFile(fileName)
+
+function processFile(n: string) {
+    fs.readFile(n, 'utf8', function (err,data) {
+        if (err) {
+          return console.log(err);
+        }
+        handleJson(JSON.parse(data))
+      });
+}
+
+function handleJson(data: any) {
+    const cluster = data["world"]["clusters"]["cluster"]
+    const selectedFields = new Array()
+    for (let m of cluster) {
+      const sf = {
+        "name": m["@displayname"],
+        "type": m["@type"],
+        "resources": extractResources(m),
+        "markers": extractMapMarkers(m),
+      }
+      selectedFields.push(sf)
+    }
+    const stringData = JSON.stringify(selectedFields.filter(isValidZone))
+    fs.writeFileSync("data-dump.json", stringData)
+}
+
+function isValidZone(sf: any) {
+  const type: String = sf["type"]
+  return (type.startsWith("TUNNEL") || type.startsWith("OPENPVP") || type.startsWith("SAFEAREA"))
+}
+
+function extractResources(m: any) {
+  if (("distribution" in m) && ("resource" in m["distribution"])) {
```

Comment:
This if could be written has `if (m.distribution?.resource)`

Reply:
Or even the whole function could return `m.distribution?.resource ?? []`

Reply:
As for the object vs list vs null. That was not something in the result, but in the source. So might have to check for that here as well.

Reply:
the object vs list item I do notice now. There are still a few scenarios where it ends up with just an object there and not a list, but it is a few less than previously

I will make the change for using ternary